### PR TITLE
Support destructuring in `for-in` loops

### PR DESCRIPTION
**Pull request by haltcase**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| Breaking change?  | no
| New feature?      | yes
| Deprecations?     | no
| Spec compliancy?  | no
| Tests added/pass? | yes
| Fixed tickets     | [#28](https://github.com/lightscript/lightscript/issues/28)
| License           | MIT

```js
obj = { dude: 'sweet' }
for val { dude } in obj:
  console.log(dude)
  // -> 'sweet'
```

```js
arr = [
  [1, 2, 3]
  [4, 5, 6]
  [7, 8, 9]
]

for elem [x, y, z] in arr:
  console.log(x, y, z)
  // -> 1, 2, 3
  // -> 4, 5, 6
  // -> 7, 8, 9
```

### CODE REVIEW

File: src/plugins/lightscript.js [hunk 1]
```diff
@@ -79,7 +79,9 @@ pp.parseForInIterationVariable = function (node, targetType = null) {
   }
 
   this.next();
-  node[iterType] = this.parseBindingIdentifier();
+  node[iterType] = iterType === 'elem' || iterType === 'val'
+    ? this.parseBindingAtom()
```
Comment (rattrayalex) [inline src/plugins/lightscript.js:83]: nice!

**Review by rattrayalex — COMMENTED** [review]
This is awesome! You sure ramped up quickly on the parser! 

The tests are great, too! Let's add one or two for "complex" destructuring, eg; 

```
for elem { a, b: { c: d }, e = 1 } in arr:
```
and also an invalid object: 
```
for val { a: 1 } in obj:
```

File: src/plugins/lightscript.js (hunk 1 above)
Comment (haltcase) [reply src/plugins/lightscript.js:83]: :tada:

**Review by rattrayalex — APPROVED** [review]
Beautiful!! Thanks!

**Pull request merged**